src/app/services/pending-transfer-polling.service.ts

```diff
@@ -75,23 +76,24 @@ export class PendingTransferPollingService {
         );
         for (const pendingTransfer of pendingTransfers) {
             let message: UiMessage;
+            const token = pendingTransfer.userToken;
+            const formattedAmount = amountToDecimal(
+                pendingTransfer.locked_amount,
+                token.decimals
+            ).toFixed();
```

Comment:
This `toFixed` comes from a `number`? If so, it may work fine now, but eventually will fail due to loss of precision on `JSON.parse`ing big numbers. Eventually, a solution like `lossless-json` with `BigNumber` may be required to show proper values

Reply:
No, the `toFixed` comes from a `BigNumber`. I already converted all numbers coming from the API to `BigNumbers` using `lossless-json`.